File: Chapter-2/2/count.kt
fun main(args: Array<String>) {
  print("What is the input string? ")
  val sentence = readLine()
  if (sentence!!.length == 0) {
    println("Empty string!")
  } else {
    println("%s has %d characters.".format(sentence, sentence.length))
  }
}

File: Chapter-2/2/count.rs
use std::io::prelude::*;
use std::io;

fn main() {
  let mut buffer = String::new();
  print!("What is the input string? ");
  io::stdout().flush().unwrap();

  io::stdin().read_line(&mut buffer).unwrap();
  let sentence = buffer.trim_right();

  let count = sentence.chars().count();
  if count == 0 {
    println!("Empty string!");
  } else {
    println!("{} has {} characters", sentence, count);
  }
}
